File: android/src/main/java/com/instabug/reactlibrary/ArgsRegistry.java
package com.instabug.reactlibrary;

import androidx.annotation.NonNull;

import com.instabug.bug.BugReporting;
import com.instabug.bug.invocation.Option;
import com.instabug.crash.models.IBGNonFatalException;
import com.instabug.featuresrequest.ActionType;
import com.instabug.library.InstabugColorTheme;
import com.instabug.library.InstabugCustomTextPlaceHolder.Key;
import com.instabug.library.OnSdkDismissCallback.DismissType;
import com.instabug.library.ReproMode;
import com.instabug.library.extendedbugreport.ExtendedBugReport;
import com.instabug.library.internal.module.InstabugLocale;
import com.instabug.library.invocation.InstabugInvocationEvent;
import com.instabug.library.invocation.util.InstabugFloatingButtonEdge;
import com.instabug.library.invocation.util.InstabugVideoRecordingButtonPosition;
import com.instabug.library.sessionreplay.model.SessionMetadata;
import com.instabug.library.ui.onboarding.WelcomeMessage;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

final class ArgsRegistry {

    static class ArgsMap<T> extends HashMap<String, T> {
        @NonNull
        @Override
        public T getOrDefault(Object key, T defaultValue) {
            final T value = get(key);
            return value != null ? value : defaultValue;
        }

        public ArrayList<T> getAll(ArrayList<String> keys) {
            final ArrayList<T> values = new ArrayList<>();
            for (String key : keys) {
                values.add(get(key));
            }
            return values;
        }
    }

    @SuppressWarnings("deprecation")
    static Map<String, Object> getAll() {
        return new HashMap<String, Object>() {{
            putAll(invocationEvents);
            putAll(invocationOptions);
            putAll(colorThemes);
            putAll(floatingButtonEdges);
            putAll(recordButtonPositions);
            putAll(welcomeMessageStates);
            putAll(reportTypes);
            putAll(dismissTypes);
            putAll(actionTypes);
            putAll(extendedBugReportStates);
            putAll(reproModes);
            putAll(sdkLogLevels);
            putAll(nonFatalExceptionLevel);
            putAll(locales);
            putAll(placeholders);
            putAll(launchType);
        }};
    }

    public static ArgsMap<IBGNonFatalException.Level> nonFatalExceptionLevel = new ArgsMap<IBGNonFatalException.Level>() {{
        put("nonFatalErrorLevelCritical", IBGNonFatalException.Level.CRITICAL);
        put("nonFatalErrorLevelError", IBGNonFatalException.Level.ERROR);
        put("nonFatalErrorLevelWarning", IBGNonFatalException.Level.WARNING);
        put("nonFatalErrorLevelInfo", IBGNonFatalException.Level.INFO);
    }};

    static ArgsMap<InstabugInvocationEvent> invocationEvents = new ArgsMap<InstabugInvocationEvent>() {{
        put("invocationEventNone", InstabugInvocationEvent.NONE);
        put("invocationEventShake", InstabugInvocationEvent.SHAKE);
        put("invocationEventFloatingButton", InstabugInvocationEvent.FLOATING_BUTTON);
        put("invocationEventScreenshot", InstabugInvocationEvent.SCREENSHOT);
        put("invocationEventTwoFingersSwipeLeft", InstabugInvocationEvent.TWO_FINGER_SWIPE_LEFT);
    }};

    static final ArgsMap<Integer> invocationOptions = new ArgsMap<Integer>() {{
        put("optionEmailFieldHidden", Option.EMAIL_FIELD_HIDDEN);
        put("optionEmailFieldOptional", Option.EMAIL_FIELD_OPTIONAL);
        put("optionCommentFieldRequired", Option.COMMENT_FIELD_REQUIRED);
        put("optionDisablePostSendingDialog", Option.DISABLE_POST_SENDING_DIALOG);
    }};

    static final ArgsMap<InstabugColorTheme> colorThemes = new ArgsMap<InstabugColorTheme>() {{
        put("colorThemeLight", InstabugColorTheme.InstabugColorThemeLight);
        put("colorThemeDark", InstabugColorTheme.InstabugColorThemeDark);
    }};

    static final ArgsMap<InstabugFloatingButtonEdge> floatingButtonEdges = new ArgsMap<InstabugFloatingButtonEdge>() {{
        put("left", InstabugFloatingButtonEdge.LEFT);
        put("right", InstabugFloatingButtonEdge.RIGHT);
        put("floatingButtonEdgeLeft", InstabugFloatingButtonEdge.LEFT);
        put("floatingButtonEdgeRight", InstabugFloatingButtonEdge.RIGHT);
        put("rectMinXEdge", InstabugFloatingButtonEdge.LEFT);
        put("rectMaxXEdge", InstabugFloatingButtonEdge.RIGHT);
    }};

    static ArgsMap<InstabugVideoRecordingButtonPosition> recordButtonPositions = new ArgsMap<InstabugVideoRecordingButtonPosition>() {{
        put("topLeft", InstabugVideoRecordingButtonPosition.TOP_LEFT);
        put("topRight", InstabugVideoRecordingButtonPosition.TOP_RIGHT);
        put("bottomLeft", InstabugVideoRecordingButtonPosition.BOTTOM_LEFT);
        put("bottomRight", InstabugVideoRecordingButtonPosition.BOTTOM_RIGHT);
    }};

    static ArgsMap<WelcomeMessage.State> welcomeMessageStates = new ArgsMap<WelcomeMessage.State>() {{
        put("welcomeMessageModeLive", WelcomeMessage.State.LIVE);
        put("welcomeMessageModeBeta", WelcomeMessage.State.BETA);
        put("welcomeMessageModeDisabled", WelcomeMessage.State.DISABLED);
    }};

    static final ArgsMap<Integer> reportTypes = new ArgsMap<Integer>() {{
        put("bugReportingReportTypeBug", BugReporting.ReportType.BUG);
        put("bugReportingReportTypeFeedback", BugReporting.ReportType.FEEDBACK);
        put("bugReportingReportTypeQuestion", BugReporting.ReportType.QUESTION);
    }};

    static final ArgsMap<DismissType> dismissTypes = new ArgsMap<DismissType>() {{
        put("dismissTypeSubmit", DismissType.SUBMIT);
        put("dismissTypeCancel", DismissType.CANCEL);
        put("dismissTypeAddAttachment", DismissType.ADD_ATTACHMENT);
    }};

    static final ArgsMap<Integer> actionTypes = new ArgsMap<Integer>() {{
        put("allActions", ActionType.REQUEST_NEW_FEATURE | ActionType.ADD_COMMENT_TO_FEATURE);
        put("requestNewFeature", ActionType.REQUEST_NEW_FEATURE);
        put("addCommentToFeature", ActionType.ADD_COMMENT_TO_FEATURE);
    }};

    static ArgsMap<ExtendedBugReport.State> extendedBugReportStates = new ArgsMap<ExtendedBugReport.State>() {{
        put("enabledWithRequiredFields", ExtendedBugReport.State.ENABLED_WITH_REQUIRED_FIELDS);
        put("enabledWithOptionalFields", ExtendedBugReport.State.ENABLED_WITH_OPTIONAL_FIELDS);
        put("disabled", ExtendedBugReport.State.DISABLED);
    }};

    static final ArgsMap<Integer> reproModes = new ArgsMap<Integer>() {{
        put("reproStepsEnabledWithNoScreenshots", ReproMode.EnableWithNoScreenshots);
        put("reproStepsEnabled", ReproMode.EnableWithScreenshots);
        put("reproStepsDisabled", ReproMode.Disable);
    }};

    static final ArgsMap<Integer> sdkLogLevels = new ArgsMap<Integer>() {{
        put("sdkDebugLogsLevelNone", com.instabug.library.LogLevel.NONE);
        put("sdkDebugLogsLevelError", com.instabug.library.LogLevel.ERROR);
        put("sdkDebugLogsLevelDebug", com.instabug.library.LogLevel.DEBUG);
        put("sdkDebugLogsLevelVerbose", com.instabug.library.LogLevel.VERBOSE);
    }};

    static final ArgsMap<InstabugLocale> locales = new ArgsMap<InstabugLocale>() {{
        put("localeArabic", InstabugLocale.ARABIC);
        put("localeAzerbaijani", InstabugLocale.AZERBAIJANI);
        put("localeChineseSimplified", InstabugLocale.SIMPLIFIED_CHINESE);
        put("localeChineseTraditional", InstabugLocale.TRADITIONAL_CHINESE);
        put("localeCzech", InstabugLocale.CZECH);
        put("localeDanish", InstabugLocale.DANISH);
        put("localeDutch", InstabugLocale.NETHERLANDS);
        put("localeEnglish", InstabugLocale.ENGLISH);
        put("localeFrench", InstabugLocale.FRENCH);
        put("localeGerman", InstabugLocale.GERMAN);
        put("localeIndonesian", InstabugLocale.INDONESIAN);
        put("localeItalian", InstabugLocale.ITALIAN);
        put("localeJapanese", InstabugLocale.JAPANESE);
        put("localeKorean", InstabugLocale.KOREAN);
        put("localeNorwegian", InstabugLocale.NORWEGIAN);
        put("localePolish", InstabugLocale.POLISH);
        put("localePortugueseBrazil", InstabugLocale.PORTUGUESE_BRAZIL);
        put("localePortuguesePortugal", InstabugLocale.PORTUGUESE_PORTUGAL);
        put("localeRomanian", InstabugLocale.ROMANIAN);
        put("localeRussian", InstabugLocale.RUSSIAN);
        put("localeSpanish", InstabugLocale.SPANISH);
        put("localeSlovak", InstabugLocale.SLOVAK);
        put("localeSwedish", InstabugLocale.SWEDISH);
        put("localeTurkish", InstabugLocale.TURKISH);
    }};

    static final ArgsMap<Key> placeholders = new ArgsMap<Key>() {{
        put("shakeHint", Key.SHAKE_HINT);
        put("swipeHint", Key.SWIPE_HINT);
        put("invalidEmailMessage", Key.INVALID_EMAIL_MESSAGE);
        put("emailFieldHint", Key.EMAIL_FIELD_HINT);
        put("commentFieldHintForBugReport", Key.COMMENT_FIELD_HINT_FOR_BUG_REPORT);
        put("commentFieldHintForFeedback", Key.COMMENT_FIELD_HINT_FOR_FEEDBACK);
        put("commentFieldHintForQuestion", Key.COMMENT_FIELD_HINT_FOR_QUESTION);
        put("invocationHeader", Key.INVOCATION_HEADER);
        put("reportQuestion", Key.REPORT_QUESTION);
        put("reportBug", Key.REPORT_BUG);
        put("reportFeedback", Key.REPORT_FEEDBACK);
        put("conversationsHeaderTitle", Key.CONVERSATIONS_LIST_TITLE);
        put("addVoiceMessage", Key.ADD_VOICE_MESSAGE);
        put("addImageFromGallery", Key.ADD_IMAGE_FROM_GALLERY);
        put("addExtraScreenshot", Key.ADD_EXTRA_SCREENSHOT);
        put("addVideoMessage", Key.ADD_VIDEO);
        put("audioRecordingPermissionDeniedMessage", Key.AUDIO_RECORDING_PERMISSION_DENIED);
        put("recordingMessageToHoldText", Key.VOICE_MESSAGE_PRESS_AND_HOLD_TO_RECORD);
        put("recordingMessageToReleaseText", Key.VOICE_MESSAGE_RELEASE_TO_ATTACH);
        put("thankYouText", Key.SUCCESS_DIALOG_HEADER);
        put("videoPressRecord", Key.VIDEO_RECORDING_FAB_BUBBLE_HINT);
        put("conversationTextFieldHint", Key.CONVERSATION_TEXT_FIELD_HINT);
        put("thankYouAlertText", Key.REPORT_SUCCESSFULLY_SENT);

        put("welcomeMessageBetaWelcomeStepTitle", Key.BETA_WELCOME_MESSAGE_WELCOME_STEP_TITLE);
        put("welcomeMessageBetaWelcomeStepContent", Key.BETA_WELCOME_MESSAGE_WELCOME_STEP_CONTENT);
        put("welcomeMessageBetaHowToReportStepTitle", Key.BETA_WELCOME_MESSAGE_HOW_TO_REPORT_STEP_TITLE);
        put("welcomeMessageBetaHowToReportStepContent", Key.BETA_WELCOME_MESSAGE_HOW_TO_REPORT_STEP_CONTENT);
        put("welcomeMessageBetaFinishStepTitle", Key.BETA_WELCOME_MESSAGE_FINISH_STEP_TITLE);
        put("welcomeMessageBetaFinishStepContent", Key.BETA_WELCOME_MESSAGE_FINISH_STEP_CONTENT);
        put("welcomeMessageLiveWelcomeStepTitle", Key.LIVE_WELCOME_MESSAGE_TITLE);
        put("welcomeMessageLiveWelcomeStepContent", Key.LIVE_WELCOME_MESSAGE_CONTENT);

        put("surveysStoreRatingThanksTitle", Key.SURVEYS_STORE_RATING_THANKS_TITLE);
        put("surveysStoreRatingThanksSubtitle", Key.SURVEYS_STORE_RATING_THANKS_SUBTITLE);

        put("reportBugDescription", Key.REPORT_BUG_DESCRIPTION);
        put("reportFeedbackDescription", Key.REPORT_FEEDBACK_DESCRIPTION);
        put("reportQuestionDescription", Key.REPORT_QUESTION_DESCRIPTION);
        put("requestFeatureDescription", Key.REQUEST_FEATURE_DESCRIPTION);

        put("discardAlertTitle", Key.REPORT_DISCARD_DIALOG_TITLE);
        put("discardAlertMessage", Key.REPORT_DISCARD_DIALOG_BODY);
        put("discardAlertStay", Key.REPORT_DISCARD_DIALOG_NEGATIVE_ACTION);
        put("discardAlertDiscard", Key.REPORT_DISCARD_DIALOG_POSITIVE_ACTION);

        put("addAttachmentButtonTitleStringName", Key.REPORT_ADD_ATTACHMENT_HEADER);

        put("reportReproStepsDisclaimerBody", Key.REPORT_REPRO_STEPS_DISCLAIMER_BODY);
        put("reportReproStepsDisclaimerLink", Key.REPORT_REPRO_STEPS_DISCLAIMER_LINK);
        put("reproStepsProgressDialogBody", Key.REPRO_STEPS_PROGRESS_DIALOG_BODY);
        put("reproStepsListHeader", Key.REPRO_STEPS_LIST_HEADER);
        put("reproStepsListDescription", Key.REPRO_STEPS_LIST_DESCRIPTION);
        put("reproStepsListEmptyStateDescription", Key.REPRO_STEPS_LIST_EMPTY_STATE_DESCRIPTION);
        put("reproStepsListItemNumberingTitle", Key.REPRO_STEPS_LIST_ITEM_NUMBERING_TITLE);
        put("okButtonTitle", Key.BUG_ATTACHMENT_DIALOG_OK_BUTTON);
        put("audio", Key.CHATS_TYPE_AUDIO);
        put("image", Key.CHATS_TYPE_IMAGE);
        put("screenRecording", Key.CHATS_TYPE_VIDEO);
        put("messagesNotificationAndOthers", Key.CHATS_MULTIPLE_MESSAGE_NOTIFICATION);
        put("team", Key.CHATS_TEAM_STRING_NAME);
        put("insufficientContentMessage", Key.COMMENT_FIELD_INSUFFICIENT_CONTENT);
    }};

    public static ArgsMap<String> launchType = new ArgsMap<String>() {{
        put("cold", SessionMetadata.LaunchType.COLD);
        put("hot",SessionMetadata.LaunchType.HOT );
        put("warm",SessionMetadata.LaunchType.WARM );
    }};
    
// Temporary workaround to be removed in future release
    public static HashMap<String,String> launchTypeReversed = new HashMap<String,String>() {{
        put(SessionMetadata.LaunchType.COLD,"cold");
        put(SessionMetadata.LaunchType.HOT,"hot" );
        put(SessionMetadata.LaunchType.WARM,"warm" );
    }};

}

File: src/utils/ArgsRegistry.ts
import { NativeInstabug } from '../native';

const NativeConstants = NativeInstabug.getConstants();

/**
 * @deprecated Pass a `LogLevel` to `debugLogsLevel` in `Instabug.init` instead.
 *
 * Verbosity level of the SDK debug logs. This has nothing to do with IBGLog,
 * and only affect the logs used to debug the SDK itself.
 */
export enum sdkDebugLogsLevel {
  sdkDebugLogsLevelVerbose = NativeConstants.sdkDebugLogsLevelVerbose,
  sdkDebugLogsLevelDebug = NativeConstants.sdkDebugLogsLevelDebug,
  sdkDebugLogsLevelError = NativeConstants.sdkDebugLogsLevelError,
  sdkDebugLogsLevelNone = NativeConstants.sdkDebugLogsLevelNone,
}

/**
 * @deprecated Pass a `LogLevel` to `debugLogsLevel` in `Instabug.init` instead.
 *
 * APM Log Level.
 */
export enum logLevel {
  none = NativeConstants.logLevelNone,
  error = NativeConstants.logLevelError,
  warning = NativeConstants.logLevelWarning,
  info = NativeConstants.logLevelInfo,
  debug = NativeConstants.logLevelDebug,
  verbose = NativeConstants.logLevelVerbose,
}

/**
 * @deprecated Use `InvocationEvent` instead.
 *
 * The event used to invoke the feedback form
 */
export enum invocationEvent {
  none = NativeConstants.invocationEventNone,
  shake = NativeConstants.invocationEventShake,
  screenshot = NativeConstants.invocationEventScreenshot,
  twoFingersSwipe = NativeConstants.invocationEventTwoFingersSwipeLeft,
  floatingButton = NativeConstants.invocationEventFloatingButton,
}

/**
 * @deprecated Use `InvocationOption` instead.
 *
 * Options added while invoking bug reporting.
 */
export enum option {
  emailFieldHidden = NativeConstants.optionEmailFieldHidden,
  emailFieldOptional = NativeConstants.optionEmailFieldOptional,
  commentFieldRequired = NativeConstants.optionCommentFieldRequired,
  disablePostSendingDialog = NativeConstants.optionDisablePostSendingDialog,
}

/**
 * @deprecated Use `ColorTheme` instead.
 *
 * The color theme of the different UI elements
 */
export enum colorTheme {
  light = NativeConstants.colorThemeLight,
  dark = NativeConstants.colorThemeDark,
}

/**
 * @deprecated Use `FloatingButtonPosition` instead.
 *
 * Floating Button edges
 */
export enum floatingButtonEdge {
  left = NativeConstants.rectMinXEdge,
  right = NativeConstants.rectMaxXEdge,
}

/**
 * @deprecated Use `RecordingButtonPosition` instead.
 *
 * NativeConstants floating buttons positions.
 */
export enum position {
  bottomRight = NativeConstants.bottomRight,
  topRight = NativeConstants.topRight,
  bottomLeft = NativeConstants.bottomLeft,
  topLeft = NativeConstants.topLeft,
}

/**
 * @deprecated Use `RecordingButtonPosition` instead.
 *
 * NativeConstants floating buttons positions.
 */
export enum IBGPosition {
  bottomRight = NativeConstants.bottomRight,
  topRight = NativeConstants.topRight,
  bottomLeft = NativeConstants.bottomLeft,
  topLeft = NativeConstants.topLeft,
}

/**
 * @deprecated Use `WelcomeMessageMode` instead.
 *
 * The welcome message mode.
 */
export enum welcomeMessageMode {
  live = NativeConstants.welcomeMessageModeLive,
  beta = NativeConstants.welcomeMessageModeBeta,
  disabled = NativeConstants.welcomeMessageModeDisabled,
}

/**
 * @deprecated Use `ReportType` instead.
 *
 * Type of the report either feedback or bug.
 */
export enum reportType {
  bug = NativeConstants.bugReportingReportTypeBug,
  feedback = NativeConstants.bugReportingReportTypeFeedback,
  question = NativeConstants.bugReportingReportTypeQuestion,
}

/**
 * @deprecated Use `DismissType` instead.
 *
 * Type of SDK dismiss
 */
export enum dismissType {
  submit = NativeConstants.dismissTypeSubmit,
  cancel = NativeConstants.dismissTypeCancel,
  addAttachment = NativeConstants.dismissTypeAddAttachment,
}

/**
 * @deprecated Use `ActionType` instead.
 *
 * NativeConstants action types.
 */
export enum actionTypes {
  allActions = NativeConstants.allActions,
  reportBug = NativeConstants.reportBugAction,
  requestNewFeature = NativeConstants.requestNewFeature,
  addCommentToFeature = NativeConstants.addCommentToFeature,
}

/**
 * @deprecated Use `ExtendedBugReportMode` instead.
 *
 * The extended bug report mode
 */
export enum extendedBugReportMode {
  enabledWithRequiredFields = NativeConstants.enabledWithRequiredFields,
  enabledWithOptionalFields = NativeConstants.enabledWithOptionalFields,
  disabled = NativeConstants.disabled,
}

/**
 * @deprecated Use `ReproStepsMode` instead.
 *
 * The user steps option.
 */
export enum reproStepsMode {
  enabled = NativeConstants.reproStepsEnabled,
  disabled = NativeConstants.reproStepsDisabled,
  enabledWithNoScreenshots = NativeConstants.reproStepsEnabledWithNoScreenshots,
}

/**
 * @deprecated Use `Locale` instead.
 *
 * The supported locales
 */
export enum locale {
  arabic = NativeConstants.localeArabic,
  azerbaijani = NativeConstants.localeAzerbaijani,
  chineseSimplified = NativeConstants.localeChineseSimplified,
  chineseTraditional = NativeConstants.localeChineseTraditional,
  czech = NativeConstants.localeCzech,
  danish = NativeConstants.localeDanish,
  dutch = NativeConstants.localeDutch,
  english = NativeConstants.localeEnglish,
  french = NativeConstants.localeFrench,
  german = NativeConstants.localeGerman,
  italian = NativeConstants.localeItalian,
  japanese = NativeConstants.localeJapanese,
  korean = NativeConstants.localeKorean,
  polish = NativeConstants.localePolish,
  portugueseBrazil = NativeConstants.localePortugueseBrazil,
  romanian = NativeConstants.localeRomanian,
  russian = NativeConstants.localeRussian,
  spanish = NativeConstants.localeSpanish,
  swedish = NativeConstants.localeSwedish,
  turkish = NativeConstants.localeTurkish,
}

/**
 * @deprecated Use `StringKey` instead.
 *
 * NativeConstants strings
 */
export enum strings {
  shakeHint = NativeConstants.shakeHint,
  swipeHint = NativeConstants.swipeHint,
  edgeSwipeStartHint = NativeConstants.edgeSwipeStartHint,
  startAlertText = NativeConstants.startAlertText,
  invalidEmailMessage = NativeConstants.invalidEmailMessage,
  invalidEmailTitle = NativeConstants.invalidEmailTitle,
  invalidCommentMessage = NativeConstants.invalidCommentMessage,
  invalidCommentTitle = NativeConstants.invalidCommentTitle,
  invocationHeader = NativeConstants.invocationHeader,
  reportQuestion = NativeConstants.reportQuestion,
  reportBug = NativeConstants.reportBug,
  reportFeedback = NativeConstants.reportFeedback,
  emailFieldHint = NativeConstants.emailFieldHint,
  commentFieldHintForBugReport = NativeConstants.commentFieldHintForBugReport,
  commentFieldHintForFeedback = NativeConstants.commentFieldHintForFeedback,
  commentFieldHintForQuestion = NativeConstants.commentFieldHintForQuestion,
  videoPressRecord = NativeConstants.videoPressRecord,
  addVideoMessage = NativeConstants.addVideoMessage,
  addVoiceMessage = NativeConstants.addVoiceMessage,
  addImageFromGallery = NativeConstants.addImageFromGallery,
  addExtraScreenshot = NativeConstants.addExtraScreenshot,
  audioRecordingPermissionDeniedTitle = NativeConstants.audioRecordingPermissionDeniedTitle,
  audioRecordingPermissionDeniedMessage = NativeConstants.audioRecordingPermissionDeniedMessage,
  microphonePermissionAlertSettingsButtonText = NativeConstants.microphonePermissionAlertSettingsButtonTitle,
  recordingMessageToHoldText = NativeConstants.recordingMessageToHoldText,
  recordingMessageToReleaseText = NativeConstants.recordingMessageToReleaseText,
  conversationsHeaderTitle = NativeConstants.conversationsHeaderTitle,
  screenshotHeaderTitle = NativeConstants.screenshotHeaderTitle,
  okButtonText = NativeConstants.okButtonTitle,
  cancelButtonText = NativeConstants.cancelButtonTitle,
  thankYouText = NativeConstants.thankYouText,
  audio = NativeConstants.audio,
  image = NativeConstants.image,
  team = NativeConstants.team,
  messagesNotification = NativeConstants.messagesNotification,
  messagesNotificationAndOthers = NativeConstants.messagesNotificationAndOthers,
  conversationTextFieldHint = NativeConstants.conversationTextFieldHint,
  collectingDataText = NativeConstants.collectingDataText,
  thankYouAlertText = NativeConstants.thankYouAlertText,
  welcomeMessageBetaWelcomeStepTitle = NativeConstants.welcomeMessageBetaWelcomeStepTitle,
  welcomeMessageBetaWelcomeStepContent = NativeConstants.welcomeMessageBetaWelcomeStepContent,
  welcomeMessageBetaHowToReportStepTitle = NativeConstants.welcomeMessageBetaHowToReportStepTitle,
  welcomeMessageBetaHowToReportStepContent = NativeConstants.welcomeMessageBetaHowToReportStepContent,
  welcomeMessageBetaFinishStepTitle = NativeConstants.welcomeMessageBetaFinishStepTitle,
  welcomeMessageBetaFinishStepContent = NativeConstants.welcomeMessageBetaFinishStepContent,
  welcomeMessageLiveWelcomeStepTitle = NativeConstants.welcomeMessageLiveWelcomeStepTitle,
  welcomeMessageLiveWelcomeStepContent = NativeConstants.welcomeMessageLiveWelcomeStepContent,
  surveysStoreRatingThanksTitle = NativeConstants.surveysStoreRatingThanksTitle,
  surveysStoreRatingThanksSubtitle = NativeConstants.surveysStoreRatingThanksSubtitle,
  reportBugDescription = NativeConstants.reportBugDescription,
  reportFeedbackDescription = NativeConstants.reportFeedbackDescription,
  reportQuestionDescription = NativeConstants.reportQuestionDescription,
  requestFeatureDescription = NativeConstants.requestFeatureDescription,
  discardAlertTitle = NativeConstants.discardAlertTitle,
  discardAlertMessage = NativeConstants.discardAlertMessage,
  discardAlertCancel = NativeConstants.discardAlertCancel,
  discardAlertAction = NativeConstants.discardAlertAction,
  addAttachmentButtonTitleStringName = NativeConstants.addAttachmentButtonTitleStringName,
  reportReproStepsDisclaimerBody = NativeConstants.reportReproStepsDisclaimerBody,
  reportReproStepsDisclaimerLink = NativeConstants.reportReproStepsDisclaimerLink,
  reproStepsProgressDialogBody = NativeConstants.reproStepsProgressDialogBody,
  reproStepsListHeader = NativeConstants.reproStepsListHeader,
  reproStepsListDescription = NativeConstants.reproStepsListDescription,
  reproStepsListEmptyStateDescription = NativeConstants.reproStepsListEmptyStateDescription,
  reproStepsListItemTitle = NativeConstants.reproStepsListItemTitle,
  screenRecording = NativeConstants.screenRecording,
  insufficientContentMessage = NativeConstants.insufficientContentMessage,
  insufficientContentTitle = NativeConstants.insufficientContentTitle,
}
